Diary/Model/DiaryManager.swift

```diff
@@ -7,12 +7,21 @@
 
 import OSLog
 
-struct DiaryManager {
+final class DiaryManager {
```

Comment:
하나하나 말하기 힘들어서 ..
정답을 주는게 야곰 스타일이 아니긴 한데 저라면 이런식으로 짤 거 같아요
이해 한되는 부분 있으면 말해주세요
```suggestion
// 필요하다면 protocol을 만들어서 준수시키고,
actor DiaryManager {
    var diaryContents: [DiaryContent]
    
    init() throws {
        // 코어데이터에 있는 일기들 가져오기
        // 가져오기 성공하면 self에 넣어주고, 아니면 throw
        self.diaryContents = // 가져온 일기들
    }
    
    func refreshDiaries() throws {
        // 전체 배열을 다시 불러올 필요가 있다면.
    }
    
    func write(diary: Diary) throws {
        // 일기 객체를 받아서
        // 코어데이터에 저장하고
        // 성공하면 메모리에 들고 있도록
    }
    
    // delete나 update같은 함수들 추가
}
```

Reply:
actor는 동시에 여러 스레드에서 실행되는 비동기 코드의 안전성을 보장합니다. 공유 자원에 대한 액세스를 자동으로 동기화하고, 데이터 레이스와 경쟁 상태를 방지합니다.
현재 저희 코드는 thread-safe 하지 않기 때문에 actor로 변경하면 좋을 것 같습니다.
공부해서 추후에 적용하도록 하겠습니다.